storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -390,53 +444,98 @@
-  (if (:code-distributor nimbus)
-    (while (and (> min-replication-count @current-replication-count)
-             (or (= -1 max-replication-wait-time)
-               (< @total-wait-time max-replication-wait-time)))
+        current-replication-count-jar (if (not (local-mode? conf))
+                                        (atom (get-blob-replication-count (master-stormjar-key storm-id) nimbus))
+                                        (atom min-replication-count))
+        current-replication-count-code (atom (get-blob-replication-count (master-stormcode-key storm-id) nimbus))
+        current-replication-count-conf (atom (get-blob-replication-count (master-stormconf-key storm-id) nimbus))
+        total-wait-time (atom 0)]
+    (if (:blob-store nimbus)
+      (while (and
+               (or (> min-replication-count @current-replication-count-jar)
+                   (> min-replication-count @current-replication-count-code)
+                   (> min-replication-count @current-replication-count-conf))
+               (or (= -1 max-replication-wait-time)
+                   (< @total-wait-time max-replication-wait-time)))
         (sleep-secs 1)
         (log-debug "waiting for desired replication to be achieved.
           min-replication-count = " min-replication-count  " max-replication-wait-time = " max-replication-wait-time
-          "current-replication-count = " @current-replication-count " total-wait-time " @total-wait-time)
+          (if (not (local-mode? conf))"current-replication-count for jar key = " @current-replication-count-jar)
+          "current-replication-count for code key = " @current-replication-count-code
+          "current-replication-count for conf key = " @current-replication-count-conf
+          " total-wait-time " @total-wait-time)
         (swap! total-wait-time inc)
-        (reset! current-replication-count  (.getReplicationCount (:code-distributor nimbus) storm-id))))
-  (if (< min-replication-count @current-replication-count)
-    (log-message "desired replication count "  min-replication-count " achieved,
-      current-replication-count" @current-replication-count)
-    (log-message "desired replication count of "  min-replication-count " not achieved but we have hit the max wait time "
-      max-replication-wait-time " so moving on with replication count = " @current-replication-count)
-    )))
-
-(defn- read-storm-topology [conf storm-id]
-  (let [stormroot (master-stormdist-root conf storm-id)]
-    (Utils/deserialize
-      (FileUtils/readFileToByteArray
-        (File. (master-stormcode-path stormroot))
-        ) StormTopology)))
+        (if (not (local-mode? conf))
+          (reset! current-replication-count-conf  (get-blob-replication-count (master-stormconf-key storm-id))))
+        (reset! current-replication-count-code  (get-blob-replication-count (master-stormcode-key storm-id)))
+        (reset! current-replication-count-jar  (get-blob-replication-count (master-stormjar-key storm-id)))))
```

Comment:
Re-posting since my previous comment was not addressed:

Here we are calling `get-blob-replication-count` three times with one argument, while the arity of the funciton is 2.  I think the missing argument here might be `nimbus`.